### DISCUSSION TITLE
How to use in Java? I need an example

### DISCUSSION THREAD

**Discussion by qeq66** [vestrel00/contacts-android#228] [Q&A] [answered]
😭

**Comment by vestrel00** [comment]
There are code samples for Java in every class/interface documentation.

For example,
- https://github.com/vestrel00/contacts-android/blob/40cf6f5a77b0c243a811cd2feeac663cc1d78649/core/src/main/java/contacts/core/Query.kt#L50-L71
- https://github.com/vestrel00/contacts-android/blob/40cf6f5a77b0c243a811cd2feeac663cc1d78649/core/src/main/java/contacts/core/BroadQuery.kt#L61-L79
- https://github.com/vestrel00/contacts-android/blob/40cf6f5a77b0c243a811cd2feeac663cc1d78649/core/src/main/java/contacts/core/Insert.kt#L60-L82
- https://github.com/vestrel00/contacts-android/blob/40cf6f5a77b0c243a811cd2feeac663cc1d78649/core/src/main/java/contacts/core/Update.kt#L65-L83
- https://github.com/vestrel00/contacts-android/blob/40cf6f5a77b0c243a811cd2feeac663cc1d78649/core/src/main/java/contacts/core/Delete.kt#L31-L37

Let me know if those are enough to get you started 😁 

In general, usage between Java and Kotlin is very similar (except for advanced queries). This is why I did not really provide Java samples in the markdown files and [documentation website](https://vestrel00.github.io/contacts-android/).

But I do see your point. I should include Java samples in the documentation website too! Here’s what I’ll do to make the documentation friendlier for Java-only users;

- [x] Ensure that all class/interface documentation have Java code samples. _This is mostly done. I just saw some without any Java samples; the APIs for SIM card and blocked numbers._
- [x] Create a “cheat sheet” page containing copy-paste-ready code library users would most likely use, in both Kotlin and Java. _I should use MkDocs extension that allows switching between Java and Kotlin for the code blocks on this particular page only._
- [x] Add a note in the cheat sheet page about why all other pages only have Kotlin samples. The reason is to keep the markdown files clean and free of non-markdown markups from MkDocs.
- [x] Add each Kotlin/Java code block in its own file in the `sample` module so that contributors get compiler errors when making breaking changes. This will prompt contributors to update the code samples and documentation!

**Comment by vestrel00** [comment, marked as answer]
@qeq66, I started working on a "cheatsheet" page that contains Kotlin AND Java copy-paste-ready sample code! It's not complete yet but you can visit the page now; https://vestrel00.github.io/contacts-android/cheatsheet/

Here's what it looks like 😁 

https://user-images.githubusercontent.com/4332753/173434151-aa08ee83-f4b5-4e75-a73d-2363b512bdc9.mov

I will continue working on the rest of the page whenever I have time.

**Comment by vestrel00** [comment]
I will convert this into a Q&A discussion 😁 
Lot's more work to do here!

**Comment by vestrel00** [comment]
The cheatsheet is complete! https://vestrel00.github.io/contacts-android/cheatsheet/

Take a look! Thanks again for this brilliant callout @qeq66 🥂

**Reply by qeq66** [reply]
This is great. It's perfect
